py/server/deephaven/experimental/iceberg.py

```diff
@@ -233,13 +499,61 @@ def table(self, instructions: Optional[IcebergReadInstructions] = None) -> Icebe
                 instructions.
 
         Returns:
-            Table: the table read from the catalog.
+            the table read from the catalog.
         """
 
-        if instructions is not None:
+        if instructions:
             return IcebergTable(self.j_object.table(instructions.j_object))
         return IcebergTable(self.j_object.table())
 
+    def table_writer(self, writer_options: TableWriterOptions):
+        """
+        Create a new IcebergTableWriter for this Iceberg table using the provided TableWriterOptions.
+        This method will perform schema validation to ensure that the provided table definition from the writer options
+        is compatible with the Iceberg table schema. All further writes performed by the returned writer will not be
+        validated against the table's schema, and thus would be faster.
+
+        Creating a table writer is the recommended approach if users want to write to the same iceberg table multiple
+        times.
+
+        Args:
+            writer_options: The options to configure the table writer.
+
+        Returns:
+            the table writer object
+        """
+        return IcebergTableWriter(self.j_object.tableWriter(writer_options.j_object))
+
+    def append(self, instructions: IcebergParquetWriteInstructions):
+        """
+        Append the provided Deephaven tables from the write instructions as new partitions to the existing Iceberg
+        table in a single snapshot. This will not change the schema of the existing table.
+
+        This method will create a new IcebergTableWriter with the provided table definition from the write instructions,
+        and use that writer to write the data to the table. Therefore, this method is not recommended if users want to
+        write to the iceberg table multiple times. Instead, users should create a single IcebergTableWriter and use it
+        to append multiple times.
+
+        Args:
+            instructions (IcebergParquetWriteInstructions): the customization instructions for write.
+        """
+        self.j_object.append(instructions.j_object)
+
+    def overwrite(self, instructions: IcebergParquetWriteInstructions):
+        """
+        Overwrite the existing Iceberg table with the provided Deephaven tables from the write instructions in a
+        single snapshot. This will delete all existing data but will not change the schema of the existing table.
+        Overwriting a table while racing with other writers can lead to failure/undefined results.
+
+        This method will create a new IcebergTableWriter with the provided table definition from the write instructions,
+        and use that writer to write the data to the table. Therefore, this method is not recommended if users want to
+        write to the iceberg table multiple times. Instead, users should create a single IcebergTableWriter and use it
+        to write multiple times.
```

Comment:
Should this specify the method to create a new IcebergTableWriter? I had to search for a bit to figure out how since the constructor should not be directly used by a user, and the proper method is missing the type hint.

Reply:
These methods are removed from this PR and will be added later if needed.